core/sail/federation/src/main/java/org/eclipse/rdf4j/sail/federation/algebra/OwnedTupleExpr.java

```diff
@@ -55,6 +55,10 @@ public void prepare(QueryLanguage queryLn, String qry, Map<String, String> bindi
 		this.variables = bindings;
 	}
 
+	public TupleQuery getQuery() {
```

Comment:
I'm a bit wary of publicly exposing too many internals. The only thing I see you do with it is retrieving it to check it's not null. Perhaps a boolean hasQuery method would be better?